tutorials/early_stopping/early_stopping.ipynb

```diff
@@ -0,0 +1,978 @@
+    "\n",
+    "We will optimize the validation accuracy, which is a `TensorboardCurveMetric` that points to the logging directory assigned above. Note that we have set `is_available_while_running`, allowing for the metric to be queried as the trial progresses. This is critical for the early stopping strategy to monitor partial results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "0775a96e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MyTensorboardMetric(TensorboardCurveMetric):\n",
+    "\n",
+    "    # NOTE: We need to tell the new Tensorboard metric how to get the id /\n",
+    "    # file handle for the tensorboard logs from a trial. In this case\n",
+    "    # our convention is to just save a separate file per trial in\n",
+    "    # the pre-specified log dir.\n",
+    "    @classmethod\n",
+    "    def get_ids_from_trials(cls, trials):\n",
+    "        return {\n",
+    "            trial.index: Path(log_dir).joinpath(str(trial.index)).as_posix()\n",
+    "            for trial in trials\n",
+    "        }\n",
+    "\n",
+    "    # This indicates whether the metric is queryable while the trial is\n",
+    "    # still running. This is required for early stopping to monitor the\n",
+    "    # progress of the running trial.ArithmeticError\n",
+    "    @classmethod\n",
+    "    def is_available_while_running(cls):\n",
+    "        return True"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "a5c5a7d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "val_acc = MyTensorboardMetric(\n",
+    "    name=\"val_acc\",\n",
+    "    curve_name=\"val_acc\",\n",
+    "    lower_is_better=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d4f3ba5d",
+   "metadata": {},
+   "source": [
+    "## Setting up the OptimizationConfig\n",
+    "\n",
+    "The `OptimizationConfig` specifies the objective for Ax to optimize."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "ada66cf3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "opt_config = OptimizationConfig(\n",
+    "    objective=Objective(\n",
+    "        metric=val_acc,\n",
+    "        minimize=False,\n",
+    "    )\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "57aa9cf7",
+   "metadata": {},
+   "source": [
+    "## Defining an Early Stopping Strategy\n",
+    "\n",
+    "A `PercentileEarlyStoppingStrategy` is a simple method that stops a trial if its performance falls below a certain percentile of other trials at the same step (e.g., when `percentile_threshold` is 50, at a given point in time, if a trial ranks in the bottom 50% of trials, it is stopped). \n",
```

Comment:
Should we mention model-based ES strategies here or is that a bit too much preview / information overload?